src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardSamUserDirectives.scala

```diff
@@ -152,4 +163,18 @@ final case class OIDCHeaders(
     email: WorkbenchEmail,
     googleSubjectIdFromAzure: Option[GoogleSubjectId],
     managedIdentityObjectId: Option[ManagedIdentityObjectId] = None
-)
+) {
+
+  // Customized toString method so that fields are labeled and we must ensure that we do not log the Bearer Token
+  override def toString: String = {
+    val extId = externalId match {
+      case Left(googleSubjectId) => s"GoogleSubjectId($googleSubjectId)"
+      case Right(azureB2CId) => s"AzureB2CId($azureB2CId)"
+    }
+    s"OIDCHeaders(" +
+      s"externalId: $extId, " +
+      s"email: $email, " +
+      s"googleSubjectIdFromAzure: $googleSubjectIdFromAzure, " +
+      s"managedIdentityObjectId: $managedIdentityObjectId)"
```

Comment:
Do we want to be logging all of these fields for every request?

Reply:
Yes, that is the goal of the ticket.